### PR TITLE
2929 add reason step

### LINKED ISSUE #2929: Add reason step to Return Items modal
## Is your feature request related to a problem? Please describe 👀

Need to add a reason and a comment for why we are returning the goods.

## Describe the solution you'd like 🎁

- A reason step in return items modal
- On ok, call the API to create the return
- Create a mock mutation to accept stock line id, quantity, reason and comment, and return a SupplierReturn invoice node.

### PR DESCRIPTION
**Pull request by lache-melvin**
Fixes #2929

# 👩🏻‍💻 What does this PR do? 
 

Adds the second step of the return items modal, and mocks the insert mutation on Ok:
Screenshot 2024-02-14 at 2 04 31 PM

# 🧪 How has/should this change been tested? 

- Open an existing/create a new Inbound Shipment
- Select some of the shipment lines
- Select the 'Return to supplier' action
- The Return Items modal opens (populated with a dummy line)
- Click Ok & Next
- See/edit the Reason and Comment inputs for the return lines
- Clicking OK calls the mutation and closes the modal

## 💌 Any notes for the reviewer?

There aren't many frontend tests beyond the common package... is this intentional? Otherwise feeling I should be wrapping tests around the hooks/rendering logic 😅

### CODE REVIEW

File: client/packages/invoices/src/InboundShipment/DetailView/modals/NewReturnItems/useDraftReturnLines.ts
Comment (lache-melvin) [inline client/packages/invoices/src/InboundShipment/DetailView/modals/NewReturnItems/useDraftReturnLines.ts:1]: Ah sorry, did a file rename here... hook changes were adding the reasonId/comment to the SupplierReturnLine and adding the saveSupplierReturn method

Comment (lache-melvin) [reply client/packages/invoices/src/InboundShipment/DetailView/modals/NewReturnItems/useDraftReturnLines.ts:1]: Though now that I'm looking at it, should an empty comment and a default reasonId be included in the generated supplier return line from the API?

File: client/packages/invoices/src/Returns/modals/NewReturnItems/useDraftReturnLines.ts
Comment (lache-melvin) [inline client/packages/invoices/src/Returns/modals/NewReturnItems/useDraftReturnLines.ts:1]: Ah sorry, did a file rename here... hook changes were adding the reasonId/comment to the SupplierReturnLine and adding the saveSupplierReturn method.

Though now that I'm looking at it, should an empty comment and a default reasonId be included in the generated supplier return line from the API?

**Review by CarlosNZ — APPROVED** [review]
Looks good, thanks. I didn't look through it too thoroughly but it'll be helpful to get it merged and we can try and find any problems with it as we use it in the upcoming issues.

**Pull request merged**